features/0587-encryption-envelope-v2/README.md

```diff
@@ -0,0 +1,95 @@
+# Aries RFC 0587: Encryption Envelope v2
+
+- Authors: [Baha A. Shaaban](mailto:baha.shaaban@securekey.com) (SecureKey Technologies Inc.), [Troy Ronda](mailto:troy.ronda@securekey.com) (SecureKey Technologies Inc.)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2021-02-10
+- Status Note:
+- Supersedes:
+- Start Date: 2021-02-10
+- Tags: [feature](/tags.md#feature)
+
+## Summary
+
+This RFC proposes that we support the definition of envelopes from [DIDComm Messaging](https://identity.foundation/didcomm-messaging/spec).
+
+## Motivation
+
+This RFC defines ciphersuites for envelopes such that we can achieve better compatability with DIDComm Messaging being specified at DIF.
+The ciphersuites defined in this RFC are a subset of the definitions in [Aries RFC 0334-jwe-envelope](https://github.com/hyperledger/aries-rfcs/tree/master/features/0334-jwe-envelope).
+
+## Encryption Algorithms
+
+[ECDH-1PU draft 03](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-03) defines the JWE structure. The following sections summarize the supported algorithms.
+
+### Curves
+
+DIDComm Messaging currently requires support for `X25519` and `P-256` and optionally allows `P-384` and `P-521`.
+This RFC additionally requires support for `P-384`.
+i.e., This RFC requires support for `X25519`, `P-256`, `P-384` and optionally allows `P-521`.
+
+- P-256 (reference in [RFC4492](https://tools.ietf.org/search/rfc4492#appendix-A))
+- P-384 (reference in [RFC4492](https://tools.ietf.org/search/rfc4492#appendix-A))
+- P-521 (reference in [RFC4492](https://tools.ietf.org/search/rfc4492#appendix-A))
+- X25519 (reference in [RFC7748](https://tools.ietf.org/html/rfc7748#section-5))
+
+### Content Encryption Algorithms
+
+DIDComm Messaging (and this RFC) requires support for both `XC20P` and `A256GCM`.
+
+- XC20P (XChaCha20Poly1305 - reference in [xchacha draft 03](https://tools.ietf.org/html/draft-irtf-cfrg-xchacha-03))
+- A256GCM (AES-GCM with a 256 bit key - reference in [RFC7518](https://tools.ietf.org/html/rfc7518#section-5.1))
+
+### Key Wrapping Algorithms
+
+DIDComm Messaing (and this RFC) requires support for `ECDH-1PU+A256KW`.
+
+- ECDH-1PU+A256KW (defined in [ECDH-1PU draft 03](https://tools.ietf.org/html/draft-madden-jose-ecdh-1pu-03#section-2.2))
+
+## Anoncrypt equivalent
+
+For situations where the envelopes should be anonymous, we use a newly minted DID rather than a different Anoncrypt mechanism.
+The newly minted DID is used once and then discarded.
+This approach matches the DIDComm Messaging mechanism.
+
+## Mime Type
+
+The mime type associated to this envelope is `application/didcomm-encrypted+json`.
```

Comment:
The DIDComm Messaging spec currently states: `application/didcomm-enc-env`
https://identity.foundation/didcomm-messaging/spec/#https

RFC 0044 currently states (for v2): `application/didcomm-encrypted+json`

Reply:
Resolved in https://github.com/decentralized-identity/didcomm-messaging/pull/168.